Conductor/Client/Extensions/ApiExtensions.cs

```diff
@@ -25,20 +24,25 @@ public class ApiExtensions
         private const string ENV_SECRET = "SECRET";
         private const int REST_CLIENT_REQUEST_TIME_OUT = 30 * 1000;
 
-        public static Configuration Configuration { get; set; }
-
-        static ApiExtensions()
-        {
-            Configuration = new Configuration(REST_CLIENT_REQUEST_TIME_OUT)
+        private static readonly Lazy<Configuration> _lazyConfiguration =
+            new Lazy<Configuration>(() => new Configuration(REST_CLIENT_REQUEST_TIME_OUT)
             {
                 BasePath = GetEnvironmentVariable(ENV_ROOT_URI),
                 AuthenticationSettings = new OrkesAuthenticationSettings(
-                GetEnvironmentVariable(ENV_KEY_ID),
-                GetEnvironmentVariable(ENV_SECRET)
-            )
-            };
+                    GetEnvironmentVariable(ENV_KEY_ID),
+                    GetEnvironmentVariable(ENV_SECRET)
+                )
+            });
+
+        private static Configuration _customConfiguration;
+
+        public static Configuration Configuration
```

Comment:
This will allow users to not rely only on Env variables. In the code they could do this:

```csharp
ApiExtensions.Configuration = new Configuration
{
    BasePath = server,
    AuthenticationSettings = new OrkesAuthenticationSettings(key, id)
};
```